### PR TITLE
Omsmith/static stateless

### PR DESCRIPTION
**Pull request by omsmith**

### CODE REVIEW

File: src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs [hunk 1]
```diff
@@ -101,36 +99,40 @@ ImmutableHashSet<ISymbol> statelessFuncs
...
-					bool hasCaptures = TryGetCaptures(
-						context,
-						argument,
-						out ImmutableArray<ISymbol> captures
-					);
-					if( !hasCaptures ) {
-						return;
+				case AnonymousMethodExpressionSyntax anonymousMethod:
+					// allow static delegates, since they can't capture non-static variables
+					if( anonymousMethod.Modifiers.Any( x => x.Kind() == SyntaxKind.StaticKeyword ) ) {
```
Comment (omsmith) [inline src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs:104]: Vote: Grab the symbol instead?

**Comment by omsmith** [comment]
static anonymous functions are a C# 9.0 feature. https://docs.microsoft.com/en-us/dotnet/csharp/language-reference/proposals/csharp-9.0/static-anonymous-functions

**Comment by omsmith** [comment]
FYI @TimothyJCowen in case you're interested in other changes

File: src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs (hunk 1 above)
Comment (j3parker) [reply src/D2L.CodeStyle.Analyzers/Immutability/StatelessFuncAnalyzer.cs:104]: 🤔 I think this is fine

**Review by j3parker — APPROVED** [review]

**Pull request merged**